darwin/cli_functions.py

```diff
@@ -380,7 +399,89 @@ def upload_data(
+                        file_tasks[file_name] = progress.add_task(f"[blue]{file_name}", total=file_total_bytes)
+
+                    progress.update(file_tasks[file_name], completed=file_bytes_sent)
+
+                for task in progress.tasks:
+                    if task.finished and len(progress.tasks) >= 5:
+                        progress.remove_task(task.id)
+
+                progress.update(upload_tasks, total=total_file_count, advance=file_advancement)
+
+            upload_manager = dataset.push(
+                files_to_exclude=files_to_exclude,
+                fps=fps,
+                as_frames=frames,
+                files_to_upload=files,
+                path=path,
+                progress_callback=upload_callback,
+            )
+        console = Console(theme=_console_theme())
+
+        console.print()
+
+        if not upload_manager.blocked_count and not upload_manager.error_count:
+            console.print(f"All {upload_manager.total_count} files have been successfully uploaded.\n", style="success")
+            return
+
+        if upload_manager.blocked_count:
+            console.print(
+                f"{upload_manager.blocked_count} out of {upload_manager.total_count} files were skipped.\n",
+                style="warning",
+            )
+
+        if upload_manager.error_count:
+            console.print(
+                f"{upload_manager.error_count} out of {upload_manager.total_count} files couldn't be uploaded because an error occurred.\n",
+                style="error",
+            )
+
+        if not verbose:
+            console.print('Re-run with "--verbose" for further details')
+            return
+
+        error_table = Table(
+            "Dataset Item ID",
+            "Filename",
+            "Remote Path",
+            "Stage",
+            "Reason",
+            show_header=True,
+            header_style="bold cyan",
+            title="Files which were not successfully uploaded",
+        )
+
+        for item in upload_manager.blocked_items:
+            error_table.add_row(str(item.dataset_item_id), item.filename, item.path, "UPLOAD_REQUEST", item.reason)
+
+        for error in upload_manager.errors:
+            for local_file in upload_manager.local_files:
+                if local_file.local_path != error.file_path:
+                    continue
+
+                for pending_item in upload_manager.pending_items:
+                    if pending_item.filename != local_file.data["filename"]:
+                        continue
+
+                    error_table.add_row(
+                        str(pending_item.dataset_item_id),
+                        pending_item.filename,
+                        pending_item.path,
+                        error.stage.name,
+                        str(error.error),
+                    )
+                    break
+
+        console.print(error_table)
     except NotFound as e:
         _error(f"No dataset with name '{e.name}'")
```

Comment:
Not directly on related to these changes, but perhaps we can add suggestions on error messages? For example this could be updated to be: 
`No dataset with name '{e.name}', use `darwin dataset remote` to see all available datasets`

Of course; this could introduce maintenance issues if commands change over time, so maybe we could centralize the error messages for slightly easier tracking

And maybe changes like this would be best tackled in their own ticket